src/expressions.md

```diff
@@ -600,6 +600,12 @@ within this loop may exit out of this loop or return control to its head.
 See [break expressions](#break-expressions) and [continue
 expressions](#continue-expressions).
 
+The return value of a `loop` expression is `!`, which can be coerced to any
+type. However, if there is a `break` expression in the loop, the return value
```

Comment:
I didn't write the implementation for the feature so I'm not sure, but I *guess* `'inner` is not diverging, and has a return value of `()`, because it actually returns.